# During 'vagrant package' operation, 'virt-sysprep' command removes existing
# SSH host keys. However, at the next boot of the packaged box, they are not
# regenerated automatically, and this breaks Vagrant provisioning as well as
# the 'vagrant ssh' command.
#
# This patch is hopefully a temporary solution to this problem until a better
# one is found. It will disable the removal of existing SSH host keys by
# the 'virt-sysprep' command. This is not advisable in a production
# environment, however can be sufficient for a temporary, test environment.
#
# See also: https://github.com/vagrant-libvirt/vagrant-libvirt/issues/759
#
# Work directory: /usr/share/rubygems-integration/all/gems/vagrant-libvirt-*

--- lib/vagrant-libvirt/action/package_domain.rb
+++ lib/vagrant-libvirt/action/package_domain.rb
@@ -38,7 +38,7 @@
           `qemu-img rebase -p -b "" #{@tmp_img}`
           # remove hw association with interface
           # working for centos with lvs default disks
-          `virt-sysprep --no-logfile --operations defaults,-ssh-userdir -a #{@tmp_img}`
+          `virt-sysprep --no-logfile --operations defaults,-ssh-userdir,-ssh-hostkeys -a #{@tmp_img}`
           Dir.chdir(@tmp_dir)
           img_size = `qemu-img info #{@tmp_img} | grep 'virtual size' | awk '{print $3;}' | tr -d 'G'`.chomp
           File.write(@tmp_dir + '/metadata.json', metadata_content(img_size))
